File: twosum/twosum.cpp
class Solution
{
public:
  vector<int> twoSum(vector<int> &nums, int target)
  {
    unordered_map<int, int> m;
    vector<int> result;
    for (int i = 0; i < nums.size(); i++)
    {
      if (m.find(nums[i]) == m.end())
      {
        m[nums[i]] = i;
      }
      else
      {
        result.push_back(m[nums[i]]);
        result.push_back(i);
        break;
      }
    }
    return result;
  }
};

File: Two Sum/TwoSum.js
/**
 * @param {number[]} nums
 * @param {number} target
 * @return {number[]}
 */
var twoSum = function (nums, target) {
  var obj = {};
  var result = [];
  nums.forEach(function (v, k) {
    if (obj[target - v]) {
      result = [obj[target - v], k]
    } else {
      obj[v] = k
    }
  })
  return result
};